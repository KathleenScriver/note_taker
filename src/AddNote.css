.add-note {
  margin: 20px;
  padding: 5px;
  border: 2px solid #00FFD8;
  background: #DDFFEE;
  text-align: center;
  border-radius: 20px;
  box-shadow: 5px 10px 10px #888888;
  padding-bottom: 50px;
}

.input-area {
  align-items: center;
}

.input-area textarea {
  margin-top: 1px;
  margin-bottom: 5px;
  height: 120px;
  width: 220px;
  border-radius: 5px;
}

.button-section {
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-flow: row || nowrap;
}

.tag-list {
  height: 30px;
  width: 100px;
  font-size: 15px;
}

.add-button {
  margin: 10px;
  margin-top: 20px;
  background-color: #769FB6;
  border-color: #538083;
  border-width: 1px;
  border-radius: 10px;
  padding: 5px;
  color: white;
  width: 100px;
  text-align: center;
  font-size: 14px;
}
