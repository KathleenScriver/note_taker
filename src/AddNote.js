import React, { Component } from 'react';
import './AddNote.css';

export default class AddNote extends Component {
  constructor(props) {
    super(props);
    this.state = {
      noteText: ''
    }
  }

  updateNote = (event) => {
    this.setState({ noteText: event.target.value });
  }

  addNote = () => {
    this.props.addNote(this.state.noteText);
    this.setState({ noteText: '' });
  }

  render() {
    return (
      <div className="add-note">
        <h1>Add a Note</h1>
        <textarea
          type='text'
          max='250'
          onChange={this.updateNote}
          value={this.state.noteText}
        />
        <button onClick={this.addNote}>Add Note</button>
      </div>
    )
  }
}
