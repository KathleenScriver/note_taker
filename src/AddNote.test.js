import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddNote from './AddNote';

describe(AddNote, () => {
  const mockAddNote = jest.fn();
  const component = shallow(
    <AddNote addNote={mockAddNote}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <AddNote />
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('contains input form', () => {
    expect(component.find('textarea')).toHaveLength(1);
    expect(component.find('button')).toHaveLength(1);
    expect(component.find('select')).toHaveLength(1);
  })

  it('calls passed in addNote function when add button clicked', () => {
    component.find('button').simulate('click');
    expect(mockAddNote).toBeCalled();
  });

  it('updates form when keys are pressed', () => {
    const updateKey = 'test';
    component.instance().updateNoteText({ target: { value: updateKey } })
    expect(component.state('noteText')).toEqual(updateKey);
  });

  it('blanks out the noteText when the button is clicked', () => {
    const updateKey = 'test';
    component.instance().updateNoteText({ target: { value: updateKey }});
    expect(component.state('noteText')).toEqual(updateKey);
    component.find('button').simulate('click');
    expect(component.state('noteText')).toHaveLength(0);
  });
})
