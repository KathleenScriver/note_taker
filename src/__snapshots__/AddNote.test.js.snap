// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote renders and matches our snapshot 1`] = `
<div
  className="add-note"
>
  <h1>
    Add a Note
  </h1>
  <textarea
    max="250"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Note
  </button>
</div>
`;
